src/app_service/service/community/service.nim

```diff
@@ -807,6 +807,9 @@ QtObject:
       # TODO improve this by only loading the data for the wanted community
       self.chatService.loadChannelGroupById(communityId)
 
+      self.events.emit(SIGNAL_COMMUNITIES_UPDATE, CommunitiesArgs(communities: @[updatedCommunity]))
```

Comment:
Ok, is it because the logic in `SIGNAL_COMMUNITIES_UPDATE` is really needed or it's because we are not smart enough when clicking on the button to realize that we already spectate it?

Reply:
I think yes. Actually, the event processed in several places. The best way is investigate all places and get some understanding how to remove it from there.